tests/bot/utils/test_redis_cache.py

```diff
@@ -42,95 +50,105 @@ async def test_set_get_item(self):
+        await self.redis.delete("internet")
+        self.assertIs(await self.redis.get("internet"), None)
+
+    async def test_contains(self):
+        """Test that we can check membership with .contains."""
+        await self.redis.set('favorite_country', "Burkina Faso")
+
+        self.assertIs(await self.redis.contains('favorite_country'), True)
+        self.assertIs(await self.redis.contains('favorite_dentist'), False)
+
+    async def test_items(self):
+        """Test that the RedisDict can be iterated."""
+        await self.redis.clear()
+
+        # Set up our test cases in the Redis cache
+        test_cases = [
+            ('favorite_turtle', 'Donatello'),
+            ('second_favorite_turtle', 'Leonardo'),
+            ('third_favorite_turtle', 'Raphael'),
+        ]
+        for key, value in test_cases:
+            await self.redis.set(key, value)
+
+        # Consume the AsyncIterator into a regular list, easier to compare that way.
+        redis_items = [item async for item in self.redis.items()]
+
+        # These sequences are probably in the same order now, but probably
+        # isn't good enough for tests. Let's not rely on .hgetall always
+        # returning things in sequence, and just sort both lists to be safe.
+        redis_items = sorted(redis_items)
+        test_cases = sorted(test_cases)
+
+        # If these are equal now, everything works fine.
+        self.assertSequenceEqual(test_cases, redis_items)
+
+    async def test_length(self):
+        """Test that we can get the correct .length from the RedisDict."""
+        await self.redis.clear()
+        await self.redis.set('one', 1)
+        await self.redis.set('two', 2)
+        await self.redis.set('three', 3)
+        self.assertEqual(await self.redis.length(), 3)
+
+        await self.redis.set('four', 4)
+        self.assertEqual(await self.redis.length(), 4)
+
+    async def test_to_dict(self):
+        """Test that the .copy method returns a workable dictionary copy."""
+        copy = await self.redis.to_dict()
+        local_copy = {key: value async for key, value in self.redis.items()}
+        self.assertIs(type(copy), dict)
+        self.assertDictEqual(copy, local_copy)
+
+    async def test_clear(self):
+        """Test that the .clear method removes the entire hash."""
+        await self.redis.clear()
+        await self.redis.set('teddy', 'with me')
+        await self.redis.set('in my dreams', 'you have a weird hat')
+        self.assertEqual(await self.redis.length(), 2)
+
+        await self.redis.clear()
+        self.assertEqual(await self.redis.length(), 0)
+
+    async def test_pop(self):
+        """Test that we can .pop an item from the RedisDict."""
+        await self.redis.clear()
+        await self.redis.set('john', 'was afraid')
+
+        self.assertEqual(await self.redis.pop('john'), 'was afraid')
+        self.assertEqual(await self.redis.pop('pete', 'breakneck'), 'breakneck')
```

Comment:
Important test; it indeed makes sure we don't fail on a non-existing key (see the comment I made on the actual method)